app/pods/publications/publication/proofs/index/controller.js

```diff
@@ -0,0 +1,206 @@
+      this.publicationSubcase.proofPrintCorrector = proofUpload.proofPrintCorrector;
+      publicationSubcaseSave = this.publicationSubcase.save();
+    }
+
+    if (proofUpload.mustUpdatePublicationStatus) {
+      yield this.publicationService.updatePublicationStatus(
+        this.publicationFlow,
+        CONSTANTS.PUBLICATION_STATUSES.PROOF_IN,
+        proofUpload.receivedDate
+      );
+
+      this.publicationSubcase.endDate = proofUpload.receivedDate;
+      publicationSubcaseSave = this.publicationSubcase.save();
+    }
+
+    yield Promise.all([
+      proofingActivitySave,
+      ...pieceSaves,
+      publicationSubcaseSave,
+    ]);
+
+    this.send('refresh');
+    this.showProofUploadModal = false;
+  }
+
+  @task
+  *saveProofRequest(proofRequest) {
+    const now = new Date();
+
+    const uploadedPieces = proofRequest.uploadedPieces;
+    const dutch = yield this.store.findRecordByUri(
+      'language',
+      CONSTANTS.LANGUAGES.NL
+    );
+
+    yield Promise.all(
+      uploadedPieces.map((piece) => {
+        piece.publicationSubcaseSourceFor = this.publicationSubcase;
+        piece.language = dutch;
+        return piece.save();
+      })
+    );
+
+    const requestActivity = this.store.createRecord('request-activity', {
+      startDate: now,
+      publicationSubcase: this.publicationSubcase,
+      usedPieces: uploadedPieces,
+    });
+    yield requestActivity.save();
+
+    const proofingActivity = this.store.createRecord('proofing-activity', {
+      startDate: now,
+      dueDate: proofRequest.proofDueDate,
+      title: proofRequest.subject,
+      subcase: this.publicationSubcase,
+      requestActivity: requestActivity,
+      usedPieces: uploadedPieces,
+      language: yield this.store.findRecordByUri(
+        'language',
+        CONSTANTS.LANGUAGES.FR
+      ),
+    });
+    yield proofingActivity.save();
+
+    const [files, outbox, mailSettings] = yield Promise.all([
+      Promise.all(uploadedPieces.mapBy('file')),
+      this.store.findRecordByUri('mail-folder', PUBLICATION_EMAIL.OUTBOX),
+      this.store.queryOne('email-notification-setting'),
+    ]);
+    const mail = yield this.store.createRecord('email', {
+      to: mailSettings.proofRequestToEmail,
+      from: mailSettings.defaultFromEmail,
+      folder: outbox,
+      attachments: files,
+      requestActivity: requestActivity,
+      subject: proofRequest.subject,
+      message: proofRequest.message,
+    });
+    yield mail.save();
+
+    // PUBLICATION-STATUS
+    yield this.publicationService.updatePublicationStatus(
+      this.publicationFlow,
+      CONSTANTS.PUBLICATION_STATUSES.PROOF_REQUESTED
+    );
+
+    this.send('refresh');
+    this.showProofRequestModal = false;
+  }
+
+  @task
+  *deleteRequest(requestActivity) {
+    const proofingActivity = yield requestActivity.proofingActivity;
+    yield proofingActivity.destroyRecord();
+
+    const mail = yield requestActivity.email;
+    yield mail?.destroyRecord();
```

Comment:
Dit gezien bij weghalen van de promises.
Er stond `if (mail)`, maar is er dan een flow waar wij geen mail aanmaken?
Niet dat ik kan zien in `saveProofRequest`